.gitmodules

```diff
@@ -152,7 +152,8 @@
 	url = https://github.com/adafruit/Adafruit_CircuitPython_RFM69.git
 [submodule "ports/esp32s2/esp-idf"]
 	path = ports/esp32s2/esp-idf
-	url = https://github.com/adafruit/esp-idf.git
+	url = https://github.com/espressif/esp-idf.git
```

Comment:
I think we will be following the idf releases directly from now on. 
This should avoid confusion and we can switch to a patched version if such a need arises.

Reply:
In testing this on my Gravitech Cucumber i get a crash to safe mode with I2C and the SGP30.  I'll try with the UM FeatherS2 to see if this reproduces.

Reply:
I was intrigued by your crash, so I ran a very simple test

```python
import board
from busio import I2C
i2c = I2C(board.SCL, board.SDA)
```
FeatherS2 this works fine.
MetroS2, and Kaluga it goes into safemode.

I used the same BME680 on all boards.
Tested my own builds and the artifacts attached to the PR.

The difference between these groups of boards that I see is
the ones that fail  use the WROVER module and the Feather
S2 (mine is a Pre-Release) uses the raw ESP32-S2.

Microdev, if you're testing using your MicroS2, also a raw ESP32-S2, right?

Reply:
This crashes for me on a debug build but the following fixes it:

```diff
diff --git a/ports/esp32s2/common-hal/busio/I2C.c b/ports/esp32s2/common-hal/busio/I2C.c
index d45d391a2..09e8ce9ca 100644
--- a/ports/esp32s2/common-hal/busio/I2C.c
+++ b/ports/esp32s2/common-hal/busio/I2C.c
@@ -94,7 +94,7 @@ void common_hal_busio_i2c_construct(busio_i2c_obj_t *self,
     if (self->semaphore == NULL) {
         mp_raise_RuntimeError(translate("Unable to create lock"));
     }
-    xSemaphoreGive(&self->semaphore);
+    xSemaphoreGive(self->semaphore);
     self->sda_pin = sda;
     self->scl_pin = scl;
     self->i2c_num = I2C_NUM_MAX;
@@ -181,7 +181,7 @@ bool common_hal_busio_i2c_has_lock(busio_i2c_obj_t *self) {
 }
 
 void common_hal_busio_i2c_unlock(busio_i2c_obj_t *self) {
-    xSemaphoreGive(&self->semaphore);
+    xSemaphoreGive(self->semaphore);
     self->has_lock = false;
 }
```

Reply:
@microDev1  That works for me on the FeatherS2 and the MetroS2. 🎉

Reply:
I wonder if this may be the root cause for the I2C/wifi problems. I have a fix for that but it doesn't make a lot of sense in that I am removing code that should be OK.

Reply:
I am confused by this change. `self->semaphore is a `StaticQueue_t` struct, not a pointer to a such a struct. `xSemaphoreGive()` and `Take()` expect a `QueueHandle_t`, which is a `void *`. `self->semaphore` is initialized by the `xSemaphoreCreateMutexStatic(&self->semaphore)`. So the removal of the `&`s seems incorrect to me.

Reply:
I have made some changes to I2C.c and .h to more or less match what my branch has.  Similar to what @dhalbert has suggested.  I'll post diffs once I get back in later today.

Reply:
That I suggested or that @microDev1 suggested?

Reply:
Looks like Skieast made the same changes and Microdev

https://github.com/adafruit/circuitpython/compare/main...skieast:update-idf-v4.3#diff-1dda8b881430b077c387870ff34ca6eebf4a74c16067b736d53aa673dc7d7b8dL93

Reply:
Could we separate the 4.3 update from the I2C changes? We can keep the buggy I2C as it is for now. I am working on the I2C stuff in https://github.com/adafruit/circuitpython/pull/4387 and https://github.com/adafruit/esp-idf/pull/1.

Reply:
> Looks like Skieast made the same changes and Microdev
> 
> [main...skieast:update-idf-v4.3diff-1dda8b881430b077c387870ff34ca6eebf4a74c16067b736d53aa673dc7d7b8dL93](https://github.com/adafruit/circuitpython/compare/main...skieast:update-idf-v4.3diff-1dda8b881430b077c387870ff34ca6eebf4a74c16067b736d53aa673dc7d7b8dL93)

I get "there isn't anything to compare" for this link.

Reply:
GitHub weirdness... If you navigate to https://github.com/skieast/circuitpython/tree/update-idf-v4.3 and click compare I think it will work.

Reply:
Yes, getting very confusing. I will start again with the new I2C.c that appears to work with 4.2. I may have been having some laptop issues that kept blowing up builds.  Rolling back windows to last week seems to have put me back to where I was

Reply:
I just realized that we are commenting on a review. I am marking this as resolved. Lets continue the conversation below. :)